.buildkite/build.sh

```diff
@@ -28,6 +28,19 @@ extra_tags=("$@")
 # in the function call.
 declare -a tags
 
+# Usage: should-push
+#
+# This function determines if the built images should be pushed up to the Docker Hub.
+# There are a few conditions:
+#   1. This must not be a local build,
+#   2. This must not be triggered by a pull request, and
+#   3. The branch being built must be master.
+should-push() {
+  test "$BUILDKITE_PIPELINE_PROVIDER" != local &&
+    test "$BUILDKITE_PULL_REQUEST" == false &&
+    test "$BUILDKITE_BRANCH" == master
+}
```

Comment:
In various buildkite plugins they've been pulling out these reusable methods like this into a separate `util.sh` file that gets included. This makes it a lot easier to track and compare this redundant code across the repos. Thoughts about pulling this out into a separate file like that, @gustavderdrache?

Reply:
We do that for some other projects where there is more logic to keep separate. Since it's just the one function, it feels a little bit like overkill in this case.